app/controllers/application_controller.rb

```diff
@@ -1,2 +1,12 @@
 class ApplicationController < ActionController::Base
-end
+    before_action :set_current_user
+  
+  private
+    def set_current_user
+      if session[:current_user]
+        @current_user = User.find(session[:current_user])
+      else
+        @current_user = nil
+      end
+    end
+  end
```

Comment:
small style thing! `private` usually isn't indented (though that's a preference thing) and the `end` should match the indentation for the thing it closes. In this case everything is just indented one extra. :)